otx/algorithms/detection/adapters/mmdet/models/detectors/custom_deformable_detr_detector.py

```diff
@@ -0,0 +1,51 @@
+"""OTX Deformable DETR Class for mmdetection detectors."""
+
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from mmdet.models.builder import DETECTORS
+from mmdet.models.detectors.deformable_detr import DeformableDETR
+
+from otx.algorithms.common.adapters.mmcv.hooks.recording_forward_hook import (
+    ActivationMapHook,
+    FeatureVectorHook,
+)
+from otx.algorithms.common.adapters.mmdeploy.utils import is_mmdeploy_enabled
+
+
+@DETECTORS.register_module()
+class CustomDeformableDETR(DeformableDETR):
+    """Custom Deformable DETR with task adapt.
+
+    Deformable DETR does not support task adapt, so it just take task_adpat argument.
+    """
+
+    def __init__(self, *args, task_adapt=None, **kwargs):
```

Comment:
We'll have to think about removing task_adapt in the future. This comment is unrelated to this PR. :)

Reply:
I really hope so